tests/src/test/scala/org/http4s/QueryParamCodecSpec.scala

```diff
@@ -103,12 +103,23 @@ trait QueryParamCodecInstances { this: Http4sSpec =>
         .map(LocalDate.ofEpochDay))
 
   implicit val ArbitraryZonedDateTime: Arbitrary[ZonedDateTime] = {
-    val zoneIds: Seq[String] = ZoneId.getAvailableZoneIds.asScala.toSeq
+    val zoneIds = {
+      val zoneIdIter = ZoneId.getAvailableZoneIds.asScala.iterator
+      // Workaround for a bug in Java 1.8:
+      //   - https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8138664
+      // See also the original issue:
+      //   - https://github.com/http4s/http4s/issues/3664
+      if (sys.props.get("java.specification.version").contains("1.8"))
+        zoneIdIter.filterNot(_ == "GMT0")
+      else
+        zoneIdIter
+    }.toSeq
+
     Arbitrary(
       for {
-        secSinceEpoch <- Gen.choose[Long](Instant.EPOCH.getEpochSecond, Instant.now.getEpochSecond)
+        instant <- Arbitrary.arbitrary[Instant] // re-use `Gen` from `Arbitrary[Instant]` here
```

Comment:
I'm not sure if re-using `Gen` from `Arbitrary[Instant]` here is the best idea. The reason why I chose to implement a specific generator here is because `ZonedDateTime` doesn't play nice with very large or very small values. 

That's why I clamped it from `Instant.EPOCH` to `Instant.now`. I'm not too sure if this is relevant, though.

Reply:
Well, to be honest, I'm not that familiar with `ZonedDateTime` to reason about it. But at least, I haven't seen any issues with neither 'far-in-the-part` nor 'far-in-the-future' values in this test, so I decided – why not?

Reply:
Ah I see. Well it looks like as long as the arbitrarily generated instants don't fall on `MIN` or `MAX`, it shouldn't be too much of a problem. I just came across this article as I was debugging earlier on.

https://stackoverflow.com/a/45044943